### PR TITLE
Implement dataset v2

### PR DESCRIPTION
**Pull request by sinclairs**
Updates the dataset service to allow version 2.0.0 of the dataset resource to be returned.

Tests have been refactored for the new functionality.

### CODE REVIEW

File: questionnaire/dataset/dataset-service.js [hunk 1]
```diff
@@ -21,42 +72,84 @@ function createDatasetService({
...
+            if (questionAnswers !== undefined) {
+                const sectionDefinition = sections[sectionId];
+                const section = createSection({sectionDefinition});
+                const attributes = section.getAttributesByData(questionAnswers);
+
+                attributes.forEach(attribute => {
+                    const existingAttribute = dataset.get(attribute.id);
+
+                    if (existingAttribute !== undefined) {
+                        mergeAttributeValues(existingAttribute, attribute);
```
Comment (armoj) [inline questionnaire/dataset/dataset-service.js:107]: is this the functionality specifically for the merging physical injuries answers from multiple pages?

File: questionnaire/dataset/dataset-service.js [hunk 2]
```diff
@@ -21,42 +72,84 @@ function createDatasetService({
...
+                    } else {
+                        const mutatedAttribute = applyGlobalAttributeDetails(
```
Comment (armoj) [inline questionnaire/dataset/dataset-service.js:109]: are the code blocks within this if/else always mutually exclusive? I must not be understanding this logic as it appears to be the `if` deals with the munging of the phy-inj stuff... and the `else` deals with the retrieval of the global meta title (also related to phy-inj.

File: questionnaire/dataset/dataset-service.js (hunk 1 above)
Comment (sinclairs) [reply questionnaire/dataset/dataset-service.js:107]: Yeah, just any attributes that might be captured across multiple pages. Only physical injuries at the moment

File: questionnaire/dataset/dataset-service.js (hunk 2 above)
Comment (sinclairs) [reply questionnaire/dataset/dataset-service.js:109]: @armoj If there is an existing attribute it'll merge them, else it'll apply any global attribute data that has been defined.

So for physical injuries, it'll get the global label initially. Then when it encounters it again it'll merge the values.

Comment (armoj) [reply questionnaire/dataset/dataset-service.js:109]: Ok makes sense. So this code programmatically assumes that if there is a "global meta title", then there will be more than 1 attribute to merge together here (e.g. exactly like the physical injuries

**Pull request merged**